homeassistant/components/light/rest.py

```diff
@@ -0,0 +1,432 @@
+                tuple(kwargs[ATTR_RGB_COLOR])
+            rgb_color_dict = json.loads(body_rgb_color_t)
+            body_on_dict.update(rgb_color_dict)
+        if ATTR_TRANSITION in kwargs:
+            body_transition_t = self._body_transition.async_render() % \
+                kwargs[ATTR_TRANSITION]
+            transition_dict = json.loads(body_transition_t)
+            body_on_dict.update(transition_dict)
+
+        body_on_t = json.dumps(body_on_dict)
+
+        try:
+            req = yield from self.set_device_state(body_on_t)
+
+            if req.status == 200:
+                self._state = True
+            else:
+                _LOGGER.error(
+                    "Can't turn on %s. Is resource/endpoint offline?",
+                    self._resource)
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.error("Error while turn on %s", self._resource)
+
+    @asyncio.coroutine
+    def async_turn_off(self, **kwargs):
+        """Turn the device off."""
+        body_off_t = self._body_off.async_render()
+
+        try:
+            req = yield from self.set_device_state(body_off_t)
+            if req.status == 200:
+                self._state = False
+            else:
+                _LOGGER.error(
+                    "Can't turn off %s. Is resource/endpoint offline?",
+                    self._resource)
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.error("Error while turn off %s", self._resource)
+
+    @asyncio.coroutine
+    def set_device_state(self, body):
+        """Send a state update to the device."""
+        websession = async_get_clientsession(self.hass)
+
+        with async_timeout.timeout(self._timeout, loop=self.hass.loop):
+            req = yield from getattr(websession, self._method)(
+                self._resource, auth=self._auth, data=bytes(body, 'utf-8'))
+            return req
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Get the current state, catching errors."""
+        try:
+            yield from self.get_device_state(self.hass)
+        except (asyncio.TimeoutError, aiohttp.ClientError):
+            _LOGGER.exception("Error while fetch data.")
+
+    @asyncio.coroutine
+    def get_device_state(self, hass):
+        """Get the latest data from REST API and update the state."""
+        websession = async_get_clientsession(hass)
+
+        with async_timeout.timeout(self._timeout, loop=hass.loop):
+            req = yield from websession.get(self._resource, auth=self._auth)
+            text = yield from req.text()
+
+        result = self._is_on_template.\
+            async_render_with_possible_json_value(text, 'None')
+        result = result.lower()
+        if result == 'true':
+            self._state = True
+        elif result == 'false':
+            self._state = False
+        else:
+            self._state = None
+
+        result = self._supported_features_template.\
```

Comment:
That's interesting, I'm not sure if there is already a platform that may change its `supported_features` at any time. Have you tried whether the front end supports this? Maybe there need changes to be made there to update the list of supported features.

Reply:
Thanks @tinloaf. I've just checked this and I can confirm that the front end does support it. If the light is turned off and on, then `supported_features` is updated and reflected in the front end.